app/uk/gov/hmrc/incometaxsessiondata/services/SessionService.scala

```diff
@@ -16,31 +16,46 @@
 
 package uk.gov.hmrc.incometaxsessiondata.services
 
+import com.mongodb.DuplicateKeyException
+import com.mongodb.client.result.{DeleteResult, InsertOneResult}
+import org.mongodb.scala.{DuplicateKeyException, MongoException, WriteConcernException, result}
+import play.api.mvc.Result
+import play.api.mvc.Results.{Conflict, InternalServerError, Ok}
 import uk.gov.hmrc.incometaxsessiondata.models.{Session, SessionData}
 import uk.gov.hmrc.incometaxsessiondata.repositories.SessionDataRepository
 
 import javax.inject.{Inject, Singleton}
 import scala.concurrent.{ExecutionContext, Future}
+import scala.reflect.runtime.universe.Try
 
 @Singleton
 class SessionService @Inject()(
                                 repository: SessionDataRepository
                               )(implicit ec: ExecutionContext) {
 
-  def get(sessionId: String): Future[Either[Throwable, Option[SessionData]]] = {
-    repository.get(sessionId) map {
+  def get(sessionId: String, internalId: String, mtditid: String): Future[Either[Throwable, Option[SessionData]]] = {
+    repository.get(sessionId, internalId, mtditid) map {
       case Some(data: Session) =>
-        Right(Some(data))
+        Right(Some(SessionData.fromSession(data)))
       case None => Right(None)
     }
   }
 
-  def set(sessionData: SessionData): Future[Boolean] = {
-    repository.set(sessionData)
+  def getByMtditid(sessionId: String): Future[Seq[Session]] = {
```

Comment:
looks like we forgot to merge one refactoring PR back in April this year;
its very confusing to pass sessionId and use it as mtdidId ? there are different things I believe ,
so worth to avoid this we introduce for: sessionId, mtdItId  a separate value types

Reply:
since we want to move onto the implementation ticket maybe it would be best to do this refactoring after?

Reply:
yeh, should be fine to do this at later stage as a part of refactoring